Extension/src/LanguageServer/settings.ts

```diff
@@ -414,337 +423,245 @@ export class CppSettings extends Settings {
-        // These strings have default values in package.json, so should never be undefined.
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        return super.Section.get<string>("vcFormat.space.aroundAssignmentOperator")!;
+    // Returns the value of a setting as a number with proper type validation and checks if value falls within the specified range.
+    private getAsNumber(settingName: string): number;
+    private getAsNumber(settingName: string, allowNull: boolean): number | null;
+    private getAsNumber(settingName: string, allowNull: boolean = false): number | null {
+        const value: any = super.Section.get<number | null>(settingName);
+        if (allowNull && value === null) {
+            return null;
+        }
+        const setting = getRawSetting("C_Cpp." + settingName);
+        if (value === null) {
+            return setting.default;
+        }
+        // Validates the value is a number and clamps it to the specified range. Allows for undefined maximum or minimum values.
+        if (isNumber(value)) {
+            if (setting.minimum !== undefined && value < setting.minimum) {
+                return setting.minimum;
+            }
+            if (setting.maximum !== undefined && value > setting.maximum) {
+                return setting.maximum;
+            }
+            return value;
+        }
+        return setting.default;
+    }
+
+    // Returns the value of a setting as an array of strings with proper type validation and checks for valid enum values.
+    private getAsArrayOfStrings(settingName: string, allowUndefinedEnums: boolean = false): string[] {
+        const value: any = super.Section.get(settingName);
+        const setting = getRawSetting("C_Cpp." + settingName);
+        if (isArrayOfString(value)) {
+            if (!allowUndefinedEnums) {
+                if (setting.items.enum !== undefined) {
+                    if (!value.every(x => this.isValidEnum(setting.items.enum, x))) {
+                        return setting.default;
+                    }
+                }
+            }
+            return value;
+        }
+        return setting.default;
     }
 
-    public get vcFormatSpacePointerReferenceAlignment(): string {
-        // These strings have default values in package.json, so should never be undefined.
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        return super.Section.get<string>("vcFormat.space.pointerReferenceAlignment")!;
+    private getAsExcludes(settingName: string): Excludes;
+    private getAsExcludes(settingName: string, allowNull: boolean): Excludes | null;
+    private getAsExcludes(settingName: string, allowNull: boolean = false): Excludes | null {
+        const value: any = super.Section.get(settingName);
+        if (allowNull && value === null) {
+            return null;
+        }
+        if (isValidMapping(value, 'string', 'boolean')) {
+            return value as Excludes;
```

Comment:
I don't think this is going to work because Excludes can be `(boolean | { when: string })`

Reply:
You probably need a special matching function for it.  And actually I just had an idea for how to do it.  Instead of passing in the key and value types as strings, pass in a lambda that does whatever validation you want.

```
export function isValidMapping(value: any, isValidKey: (key: any) => boolean, isValidValue: (value: any) => boolean): value is object {
    if (isObject(value)) {
        return Object.entries(value).every(([key, val]) => isValidKey(key) && isValidValue(value));
    }
    return false;
}

```